include/ada/checkers.h

```diff
@@ -37,29 +36,29 @@ namespace ada::checkers {
   }
 
   // Check whether x is an ASCII digit. More likely to be inlined than std::isdigit.
-  constexpr bool is_digit(char x) { return (x >= '0') & (x <= '9'); }
+  constexpr bool is_digit(char x) noexcept { return (x >= '0') & (x <= '9'); }
 
   // A Windows drive letter is two code points, of which the first is an ASCII alpha
   // and the second is either U+003A (:) or U+007C (|).
-  inline bool is_windows_drive_letter(const std::string_view input) noexcept {
+  inline constexpr bool is_windows_drive_letter(std::string_view input) noexcept {
     return input.size() >= 2 && (is_alpha(input[0]) & ((input[1] == ':') | (input[1] == '|')));
   }
 
   // A normalized Windows drive letter is a Windows drive letter of which the second code point is U+003A (:).
-  inline bool is_normalized_windows_drive_letter(std::string_view input) noexcept {
+  inline constexpr bool is_normalized_windows_drive_letter(std::string_view input) noexcept {
     return input.size() >= 2 && (is_alpha(input[0]) & (input[1] == ':'));
   }
 
   ada_really_inline constexpr bool is_next_equals(const std::string_view::iterator start,
                                                   const std::string_view::iterator end,
-                                                  const char c) {
-    return (std::distance(start, end) > 0) && (start[1] == c);
+                                                  const char c) noexcept {
+    return (start != end) && (start[1] == c);
   }
 
   ada_really_inline constexpr bool is_not_next_equals(const std::string_view::iterator start,
                                                       const std::string_view::iterator end,
-                                                      const char c) {
-    return (std::distance(start, end) > 0) && (start[1] != c);
+                                                      const char c) noexcept {
+    return (start == end) || (start[1] != c);
```

Comment:
This here changed semantics (is that intentional), and additionally the `start[1]` read seems to be a potential OOB read regardless